src/backend/cuda/subconfigure.m4

```diff
@@ -61,51 +61,55 @@ if test "$with_cuda" != "no" ; then
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
         AC_MSG_CHECKING([whether nvcc works])
-        cat>conftest.cu<<EOF
-        __global__ void foo(int x) {}
-EOF
         if test -n "$ac_save_CC" ; then
-            NVCC_FLAGS="-ccbin $ac_save_CC"
+            NVCC_FLAGS="$NVCC_FLAGS -ccbin $ac_save_CC"
             # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
             # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
             NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
-        else
-            NVCC_FLAGS=""
         fi
         # try nvcc from PATH if 'with-cuda' does not contain a valid path
         if test -d ${with_cuda} ; then
             nvcc_bin=${with_cuda}/bin/nvcc
         else
             nvcc_bin=nvcc
         fi
-        ${nvcc_bin} $NVCC_FLAGS -c conftest.cu 2> /dev/null
-        if test "$?" = "0" ; then
+
+        # save language settings, customize ac_ext and ac_compile to support CUDA
+        AC_LANG_PUSH([C])
+        ac_ext=cu
+        ac_compile='$nvcc_bin $NVCC_FLAGS -c conftest.$ac_ext >&5'
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([__global__ void foo(int x) {}],[])],
+        [
             AC_DEFINE([HAVE_CUDA],[1],[Define is CUDA is available])
-            AS_IF([test -n "${with_cuda}"],[NVCC=${with_cuda}/bin/nvcc],[NVCC=nvcc])
+            AS_IF([test -d "${with_cuda}"],[NVCC=${with_cuda}/bin/nvcc],[NVCC=nvcc])
             AC_SUBST(NVCC)
             AC_SUBST(NVCC_FLAGS)
-            # nvcc compiled applications need libstdc++ to be able to link
-            # with a C compiler
-            PAC_PUSH_FLAG([LIBS])
+            AC_MSG_RESULT([yes])
+        ],[
+            have_cuda=no
+            AC_MSG_RESULT([no])
+            AC_MSG_ERROR([CUDA was not functional with provided host compiler (CC)])
+        ])
+        # done with CUDA, back to C
+        AC_LANG_POP([C])
+
+        # nvcc compiled applications need libstdc++ to be able to link
+        # with a C compiler
+        AC_MSG_CHECKING([if we can link libstdc++])
```

Comment:
```suggestion
        AC_MSG_CHECKING([if nvcc can link libstdc++])
```

Reply:
I guess the more correct thing is `if $CC can link libstdc++`? I believe what yaksa does is link together CUDA and C object files using the C compiler with `-lstdc++`. Maybe we should actually be linking with a C++ compiler? That addition seemed fishy to me at the time, but it seems to work 🤷.

Reply:
Ahh, that makes sense.

Reply:
I suspect from quick google that we might want to switch to linking with `nvcc`. Libtool appears to have support for it.

Reply:
I am more leaning toward linking with an actual nvcc compiled object. We need test what we are going to do.

Reply:
Yeah, I think what I am getting at is we shouldn't need to manually add libstdc++. From what I can tell, CUDA programming guides don't suggest it. My guess is as long as we link with `nvcc`, the right dependencies will be pulled in. The C compiler doesn't know to bring in any C++ libraries at link time.